vignettes/stan-help.Rmd

```diff
@@ -0,0 +1,403 @@
+
+# Failures modes
+
+In this section we will discuss the most common issues when working with `{epinowcast}` specifically and CmdStan/ Bayesian inference more generally.
+The solutions to these issues typically require you to investigate your [sampler settings](#samplerParams), [model settings](#modelParams), or [interogate your data](#modelData) further.
+
+## Sampler settings {.samplerParams}
+
+`{epinowcast}` provides users the ability to pass arguments directly to CmdStan which can improve model run time and produce better inferences.
+These parameters can be passed the `enw_fit_opts` function.
+
+### `threads_per_chain` and `chains`
+
+One of the key components of Bayesian inference is creating draws from independent Markov chains.
+To assess convergence and the reliability of a given posterior distribution it is vital that at least two chains are used with **`4` chains** being the more common default.
+
+Stan allows for multi-threading and the Stan code used within `{epinowcast}` has tried to take advantage of these capabilities where possible.
+Multi-theading allows for the calculations to be spread across multiple threads, which could accelerate inference by allowing embarassingly parallel calculations to be spread across threads and then later combined.
+If possible, a good default is `2`, however, if you have available compute on your machine, you can increase this to a higher value and may see some fit time reductions.
+Please note that the `theads_per_chain` has not impact on model convergence or fittings and is purely a means of speeding up the model fitting time.
+
+::: {.alert .alert-warning}
+**The product of the `thread_per_chain` and `chains`** arguments should be equal to or less than the number of cores on your machine! 
+:::
+
+### iter_warmup and iter_sampling {.iters}
+
+The number of samples using for the warmup phase of Stan and the number of sampling iterations are controlled by the `iter_warmup` and `iter_sampling` arguments, respectively.
+Stan requires a sufficient number of warmup iterations to dynamically estimate the step change sizes during model fitting. 
+Unlike with Gibbs sampling (such as the method used in JAGS), you do not need to set large values for `iter_warmup` or `iter_sampling`.
+Conversely, setting large values for iterations is a sign of poor model fit and indicates that other approaches need to be employed (e.g., `priors`, `adapt_delta`, `max_treedepth`)
+Typically, 1000-2000 samples for `iter_warmup` and `iter_sampling` are sufficient.
+
+### max_treedepth {.maxTree}
+
+This value controls the size of the trajectory taken by the sample (in power of 2) take for each step.
+If this value is too low, the sampler may terminate too early causing excess runtimes.
+You can increase the `max_treedepth` to an integer value greater than `10` up to `15` (or lower).
+Given the complexity of the models used in `{epinowcast}` it is common to have the `max_treedepth` set between 12 and 15.
```

Comment:
It's probably worth (1) talking about this bit in the "My model takes too long to run" section below and (2) Discussing sampler efficiency vs. bias? My understanding of treedepth warnings is that they don't necessarily represent model validity concerns, just inefficient numerical trajectories from the NUTS sampler.